uavcan/file/406.ListDirectory.0.1.uavcan

```diff
@@ -0,0 +1,36 @@
+#
+# This service can be used to list a remote directory, one entry per request.
+#
+# The client should query each entry independently, iterating 'entry_index' from 0 until the last entry.
+# When the index reaches the number of elements in the directory, the server will report that there is
+# no such entry by returning an empty name.
+#
+# The field entry_index must be applied to an ordered list of directory entries (e.g. alphabetically ordered).
+# The exact sorting criteria does not matter as long as it provides the same ordering for subsequent service calls.
+#
+# Observe that this listing operation is fundamentally non-atomic. The caller shall beware of possible race conditions
+# and is responsible for handling them properly. Particularly, consider what happens if a new item is inserted into
+# the directory between two subsequent calls: if the item happened to be inserted at the index that is lower than the
+# index of the next request, the next returned item (or several, if more items were inserted) will repeat the ones
+# that were listed earlier. The caller should handle that properly, either by ignoring the repeated items or by
+# restarting the listing operation from the beginning (index 0).
+#
```

Comment:
Too complex, doesn't provide much value: even if the snapshot is fully consistent within itself, by the time it is delivered to the client it might be obsolete.